lib/ProductOpener/Import.pm

```diff
@@ -519,6 +519,15 @@ sub set_field_value (
 			$product_ref->{$field} = "";
 			delete $product_ref->{$field . "_tags"};
 		}
+
+		# If we are on the producers platform, replace existing values by producer supplied values for allergens and traces
+		if (defined $options{replace_existing_values_when_importing_those_tags_fields}->{$field}) {
```

Comment:
I think you don't need the ->

One thing is that this code will fail of $options{replace_existing_values_when_importing_those_tags_fields] does not exist (for instance in Config_obf.pm).

So instead you can add this import at the top of the file:

use Data::DeepAccess qw(deep_exists);

and then use this condition

if (deep_exists(\%options, "replace_existing_values_when_importing_those_tags_fields", $field))

this function checks that each of the fields exist in a nested hash like %options